### DISCUSSION TITLE
How to map subpath to locale without using rewrites ?

### DISCUSSION THREAD

**Discussion by kristijorgji** [vercel/next.js#28887] [Help]
I am using next 11 and its native i18n + https://github.com/isaachinman/next-i18next

I have these configurations:

next-i18next.config.js
```
const path = require('path');

module.exports = {
    localePath: path.resolve('./public/static/locales'),
    i18n: {
        defaultLocale: 'en_US',
        locales: ['en_US', 'de_DE'],
    },
};
```

next.config.js
```
const { i18n } = require('./next-i18next.config');

/** @type {import('next/dist/server/config').NextConfig} */
module.exports = {
    i18n,
};
```

Changing locale works fine with subpaths domain.com/de_DE en_US

I would like a simple not expensive way to have subpaths like /de and /en, then in next router/request to get locale de_DE or en_US.

Is there someplace to specify such subpath to locale mapping  ?
Example:
```
{
en: en_US
de: de_DE
}
```
then domain.com/en/login will have locale en_US in all request/router places for `locale` variable
domain.com/de/login will have locale de_DE in all request/router places for `locale` variable

**Comment by apennell** [comment]
I'm trying to figure out how to do the same thing. I was able to achieve this using `localeSubpaths` with v7 of [next-i18next](https://github.com/isaachinman/next-i18next), but v8 defers to NextJS's built-in i18n routing and I haven't found a way to do this yet.
